internal/controller/monovertexrollout/monovertexrollout_controller.go

```diff
@@ -903,5 +899,12 @@ func (r *MonoVertexRolloutReconciler) SetCurrentRiderList(
 		}
 	}
 	numaLogger.Debugf("setting MonoVertexRollout.Status.Riders=%+v", monoVertexRollout.Status.Riders)
+}
 
+func (r *MonoVertexRolloutReconciler) updateProgressiveResultsMetrics(monoVertexRollout *apiv1.MonoVertexRollout, newMonoVertexDef *unstructured.Unstructured) {
+	if newMonoVertexDef != nil {
+		r.customMetrics.IncMonovertexProgressiveResults(monoVertexRollout.GetRolloutObjectMeta().GetNamespace(), monoVertexRollout.GetRolloutObjectMeta().GetName(),
+			newMonoVertexDef.GetName(), progressive.EvaluateSuccessStatusForMetrics(monoVertexRollout.GetUpgradingChildStatus().AssessmentResult),
+			string(monoVertexRollout.GetUpgradingChildStatus().BasicAssessmentResult), monoVertexRollout.GetUpgradingChildStatus().ForcedSuccess, true)
+	}
```

Comment:
Are you just keeping either this function or the one below, and still deciding which?